cadc-registry/src/test/java/ca/nrc/cadc/reg/client/RegistryClientTest.java

```diff
@@ -342,54 +311,38 @@ public void testGetServiceURLMatchDomain() {
-                                        "/" + RegistryClient.CONFIG_CACHE_DIR + "/registry-entries",
-                                capSourceCacheFile.getAbsolutePath());
-        } catch (Exception t) {
-            log.error("unexpected exception", t);
-            throw t;
-        } finally {
-            // reset
-            System.setProperty("java.io.tmpdir", currentTmpDir);
-        }
-    }
-
-    @Test
-    public void testGetCapabilitiesURL() throws Exception {
-        final String currentTmpDir = System.getProperty("java.io.tmpdir");
-        final String fileSeparator = System.getProperty("file.separator");
-        System.setProperty("java.io.tmpdir", currentTmpDir + (currentTmpDir.endsWith(fileSeparator) ? "" :
-                fileSeparator) + "build" + fileSeparator + "tmp");
-
-        try {
-            final RegistryClient registryClient =
-                    new RegistryClient(new URL("https://www.mysite.com/registry/registry-entries"));
-            final File cacheFile =
-                    new File(new File(System.getProperty("java.io.tmpdir") + fileSeparator
-                                              + System.getProperty("user.name") + fileSeparator + "cadc-registry-1.4"),
-                             "registry-entries");
-            final FileWriter fileWriter = new FileWriter(cacheFile);
+            RegistryClient registryClient = new RegistryClient();
+            String cache = RegistryClient.Query.CAPABILITIES.getValue();
+            File cacheDir = new File(System.getProperty("java.io.tmpdir") + fileSeparator
+                                              + System.getProperty("user.name") + fileSeparator + "cadc-registry-1.4");
+            cacheDir.mkdirs();
+            File cacheFile = new File(cacheDir, cache);
+            
+            FileWriter fileWriter = new FileWriter(cacheFile);
             fileWriter.write("ivo://cadc.nrc.ca/myservice/entry = https://mysite.com/services/mygreatservice");
             fileWriter.close();
 
-            final URL capabilitiesURL =
-                    registryClient.getAccessURL(URI.create("ivo://cadc.nrc.ca/myservice/entry"));
+            URI resourceID = URI.create("ivo://cadc.nrc.ca/myservice/entry");
+            URL capabilitiesURL = registryClient.getAccessURL(resourceID);
 
-            Assert.assertEquals("Wrong URL.",
+            Assert.assertEquals("default lookup",
                                 "https://mysite.com/services/mygreatservice",
                                 capabilitiesURL.toExternalForm());
+            
+            URL explicitCapURL = registryClient.getAccessURL(RegistryClient.Query.CAPABILITIES, resourceID);
+            Assert.assertEquals("explicit lookup", capabilitiesURL, explicitCapURL);
+            
+            try {
+                registryClient.getAccessURL(RegistryClient.Query.APPLICATIONS, resourceID);
```

Comment:
I see.. it looks like this test is designed to fail. (no applications cache file written above, only capabilities.) Why? Is it making sure the behaviour of getAccessURL has changed?  

There's no need to have an explicit positive test for RegistryClient.Query.APPLICATIONS?

Reply:
yes, supposed to fail becauyse wrong query; adding an Assert.fail to make test correct.

also adding a positive applications test